src/validation.cpp

```diff
@@ -2642,7 +2647,8 @@ bool CChainState::ActivateBestChain(BlockValidationState& state, std::shared_ptr
 
         {
             LOCK(cs_main);
-            LOCK(m_mempool.cs); // Lock transaction pool for at least as long as it takes for connectTrace to be consumed
+            // Lock transaction pool for at least as long as it takes for connectTrace to be consumed
+            if (m_mempool) LOCK(m_mempool->cs);
```

Comment:
Nice try, but leaving out the `{}` won't prevent the lock from being closed in the same line.

Reply:
Oh yeah d'oh. Sure could've used those missing annotations there.